### ISSUE TITLE
[data grid] Lazy loading, unstable_replaceRows does not show new line 

### ISSUE THREAD

**Issue by r-engel** [mui/mui-x#7070]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:
https://codesandbox.io/s/naughty-leftpad-7d03k6?file=/demo.js

Steps:
1. Wait till data is loaded
2. Click the click me button
3. The triggered function uses the current loaded data and adds a new item at index 1 and creates a slice to use it with  unstable_replaceRows, to simulate a new row insertion.
4. The ui tries to render the new data (flickering) and then renders the old data without the new line.

### Current behavior 😯

It seams as unstable_replaceRows may not work if there is an insertion of an new row? 

### Expected behavior 🤔

The new data provided should be set at the position and rendered, and I think it would be best to clear all rowdata that is not shown at that moment. 

### Context 🔦

My grid should be updated from wesocket messages, e.g. insert new row:
in that case the data reloads the slice currently shown to the user. and if the new item is part of the slice its shown to the user.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID 💳 (optional)

52843

**Comment by r-engel** [comment]
Hi @m4theushw, thanks for checking the issue. 
Will I get this fix for the v5 version soon?

**Comment by MBilalShafi** [comment]
Already fixed in #7694
